### PR TITLE
reloc: Remove "build_reloc.sh" script as obsolete

### PR DESCRIPTION
**Pull request by penberg**
The "ninja dist-server-tar" command is a full replacement for
"build_reloc.sh" script. We release engineering infrastructure has been
switched to ninja, so let's remove "build_reloc.sh" as obsolete.

### CODE REVIEW

**Comment by avikivity** [comment]
build_rpm.sh and build_deb.sh still refer to it, they should be adjusted.

**Comment by avikivity** [comment]
(as does the docker README)

**Comment by penberg** [comment]
@avikivity Fixed.

File: dist/docker/redhat/README.md [hunk 1]
```diff
@@ -31,9 +31,9 @@ Do the following in the top-level Scylla source directory:
 
 1. Build your own Scylla in whatever build mode you prefer, e.g., dev.
 
-2. Run `./reloc/build_reloc.sh --mode dev`
+2. Run `ninja build/dev/dist/tar/scylla-package.tar.gz`
 
-3. Run `./reloc/build_rpm.sh --reloc-pkg build/dev/scylla-package.tar.gz`
+3. Run `./reloc/build_rpm.sh --reloc-pkg ninja build/dev/dist/tar/scylla-package.tar.gz`
```
Comment (avikivity) [inline dist/docker/redhat/README.md:36]: This seems unlikely to work.

Comment (avikivity) [reply dist/docker/redhat/README.md:36]: btw, better to put the instructions for building rpm with ninja where.

Comment (penberg) [reply dist/docker/redhat/README.md]: Fixed.

**Pull request merged**